src/rnn.c

```diff
@@ -84,30 +84,199 @@ void compute_dense(const DenseLayer *layer, float *output, const float *input)
    M = layer->nb_inputs;
    N = layer->nb_neurons;
    stride = N;
-   for (i=0;i<N;i++)
+   for (i = 0;i < N;i++)
    {
       /* Compute update gate. */
       float sum = layer->bias[i];
-      for (j=0;j<M;j++)
+      for (j = 0; j<M;j++)
          sum += layer->input_weights[j*stride + i]*input[j];
       output[i] = WEIGHTS_SCALE*sum;
    }
    if (layer->activation == ACTIVATION_SIGMOID) {
-      for (i=0;i<N;i++)
+      for (i = 0;i < N;i++)
          output[i] = sigmoid_approx(output[i]);
    } else if (layer->activation == ACTIVATION_TANH) {
-      for (i=0;i<N;i++)
+      for (i = 0;i < N;i++)
          output[i] = tansig_approx(output[i]);
    } else if (layer->activation == ACTIVATION_RELU) {
-      for (i=0;i<N;i++)
+      for (i = 0;i < N;i++)
          output[i] = relu(output[i]);
    } else {
      *(int*)0=0;
    }
 }
 
+// FMA is always available if AVX2 is available
+#if !defined(__FMA__) && defined(__AVX2__)
```

Comment:
This assumption is not true. I am in a good position to attest to it, because on my machine AVX2 is supported but FMA is not, and the build fails.

Reply:
Ah, I shouldn't have believed that one random post that told me that then.

I've replaced this with a proper fallback to perform the FMA with a multiply + add if `__FMA__` isn't available.

Reply:
I have seen the same before, I guess this assumption is safe 99% of the time, but that's not good enough, is it ? 😉